packages/next/src/server/app-render/app-render.tsx

```diff
@@ -522,12 +522,29 @@ async function generateDynamicFlightRenderResult(
     onFlightDataRenderError
   )
 
-  const rscPayload = await generateDynamicRSCPayload(ctx, options)
+  const RSCPayload = await generateDynamicRSCPayload(ctx, options)
+
+  if (
+    process.env.NODE_ENV === 'development' &&
+    renderOpts.experimental.dynamicIO
```

Comment:
In the SSR path I also gate on renderOpts.dev. Someday we may want to support running next build/ next start with NODE_ENV === 'development' because it can for instance give you additional debug info. But this feature we're building is really a feature of `next dev` itself not just that we're compiled with this NODE_ENV. the reason I also use NODE_ENV is b/c the necessary debug info isn't even available in the prod builds of React so it sort of needs a double opt in.

I do think that having to read this from renderOpts is not great though and there are plenty of places in Next where we use NODE_ENV==="development" as a standin for `next dev` so we should long term move to a global like an envvar rather than piping it through as normal arguments to various functions.

Reply:
> Someday we may want to support running next build/ next start with NODE_ENV === 'development'

In fact, we already offer that option today: https://github.com/vercel/next.js/blob/3a9a50c2f119a32473d624aa32d2b28aef5445b5/packages/next/src/server/config-shared.ts#L516:L519